### ISSUE TITLE
[core] gRPC error ENHANCE_YOUR_CALM

### ISSUE THREAD

**Issue by stephanie-wang** [ray-project/ray#25367]
### What happened + What you expected to happen

During 1TB nightly tests (name dataset_shuffle*), I see error messages like this:
```
(map pid=516, ip=172.31.64.223) E0526 12:32:19.203322360     675 chttp2_transport.cc:1103]   Received a GOAWAY with error code ENHANCE_YOUR_CALM and debug data equal to "too_many_pings"
```
This could mean we're not setting some config parameters correctly/inefficiently in gRPC.

### Versions / Dependencies

2.0dev

### Reproduction script

dataset_shuffle_* nightly tests. Seems like it's shown up in other workloads too.

**Comment by stephanie-wang** [comment]
Upgrading this to P1 since the error messages are pretty scary.

**Comment by rickyyx** [comment]
> This could mean we're not setting some config parameters correctly/inefficiently in gRPC.

Yeah, I agree - I think it's essentially the combination of below configs on server and client according to this [grpc doc](https://github.com/grpc/grpc/blob/HEAD/doc/keepalive.md#faq)
- GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS
- GRPC_ARG_KEEPALIVE_TIME_MS
- GRPC_ARG_HTTP2_MIN_RECV_PING_INTERVAL_WITHOUT_DATA_MS

However, I am struggling to ping down the exact server/client, do you have any idea on which grpc servers are raising this message? @stephanie-wang 
I saw the `map` name in the log file which I assume it's a worker? E.g. this [channel](https://github1s.com/ray-project/ray/blob/HEAD/python/ray/util/client/worker.py#L188-L189) on the client definitely seems to have some non-conventional settings

**Comment by scv119** [comment]
Maybe use `GRPC_VERBOSITY=DEBUG` to see server side error log as well?

**Comment by scv119** [comment]
also did a quick search: https://sourcegraph.com/github.com/ray-project/ray/-/blob/src/ray/rpc/grpc_server.cc?L60
https://grpc.github.io/grpc/core/md_doc_keepalive.html 

maybe we set the wrong number for grpc_server?

**Comment by rkooo567** [comment]
I think we need to carefully choose these values after some research. Actually @iycheng and I decided to discuss some of gRPC config values more carefully after the rc1 branch cut.

@scv119 is this still P0? Do you think we can do this after the branch cut? Also, @rickyyx is this easy to reproduce?

**Comment by rickyyx** [comment]
> Maybe use `GRPC_VERBOSITY=DEBUG` to see server side error log as well?

Right - I tried that, but unfortunately the server side logging doesn't reveal much useful info. Tried `GRPC_TRACE` on some relevant topics as well. 

> Also, @rickyyx is this easy to reproduce?

This is fairly easy to reproduce on large shuffle's map phase. (takes maybe 10min to set up and loaded the dataset before map phase starts). Most of our e2e shuffle of 1tb will surface this. 


> maybe we set the wrong number for grpc_server?

I am fairly confident this is a client side configs issue, or some mismatch in configs between client/server.
